### PR TITLE
chore(main): Improve nightly test reports

### PR DESCRIPTION
**Pull request by SandeepTuniki**
This is a small PR to improve the debugging workflow for nightly test failures.

### Current Process

- Nightly CI runs are configured to raise an issue in the below image format when they fail.
- To find the cause of a failure, we must manually go to the Actions tab and scroll through all the CI runs. 

6g8qhJnhDn85rh6

### Improvement

- This PR improves the process by automatically linking the issue to the failing CI run. This will make it easier for us to quickly find the cause of a failure.

6RmSWyoVnVoQPW4

Tested and verified with #21478

### CODE REVIEW

File: .toys/ci.rb [hunk 1]
```diff
@@ -433,6 +437,12 @@ def encode_str str
   
   def create_body dir, tasks
     now = Time.now.utc.strftime "%Y-%m-%d %H:%M:%S"
-    "At #{now} UTC, detected failures in #{dir} for: #{tasks.join ', '}"
+    if github_action_id
```
Comment (dazuma) [inline .toys/ci.rb:440]: Currently, `github_action_id` is set to the empty string if the `--github-action-id` flag is omitted (see line 374 where you set the default to the empty string). So this if statement always succeeds (and is thus useless). However, that means the `action_url` will be set to a URL without an action ID if the `--github-action-id` flag is omitted, which is probably not what you want. Maybe consider removing the default (which will cause `github_action_id` to be nil if the flag is omitted) and then include appropriate logic to detect that condition and omit the "The CI logs can be found here" line in that case.

**Review by dazuma — CHANGES_REQUESTED** [review]

**Comment by dazuma** [comment]
I really like this change, though. It will be really helpful for troubleshooting failures.

File: .toys/ci.rb (hunk 1 above)
Comment (SandeepTuniki) [reply .toys/ci.rb]: Thanks for spotting this. I assumed empty string evaluates to false. 🤦‍♂️

I removed the default value and added a check for `nil` value.

File: .toys/ci.rb [hunk 2]
```diff
@@ -433,6 +436,11 @@ def encode_str str
   
   def create_body dir, tasks
     now = Time.now.utc.strftime "%Y-%m-%d %H:%M:%S"
-    "At #{now} UTC, detected failures in #{dir} for: #{tasks.join ', '}"
+    messages = []
+    messages << "At #{now} UTC, detected failures in #{dir} for: #{tasks.join ', '}."
+    unless github_action_id.nil?
+      messages << "The CI logs can be found [here](https://github.com/googleapis/google-cloud-ruby/actions/runs/#{github_action_id})"
+    end
+    messages.join("\n\n")
```
Comment (dazuma) [inline .toys/ci.rb:444]: Style nit: Drop the parens

**Review by dazuma — APPROVED** [review]

**Pull request merged**